gnome3_22: call for testing

groxxda opened the pull request:
**There are many updates that do not only affect gnome3.22 in this PR (at the top). Can we merge them individually? Some may already be in staging, or part of an open PR.**

~~I did not succeed in updating gdm (yet), so I kept it at 3.14 :disappointed: When I find time I'll look into it again.~~ 
gdm 3.22 works [more infos](https://github.com/NixOS/nixpkgs/pull/19081#issuecomment-251256745)

groxxda committed:
NetworkManager-strongswan: init at 1.4.0

groxxda committed:
at-spi2-{core,atk}: 2.20.1 -> 2.22.0

groxxda committed:
libsoup: libgnome_keyring -> libgnome_keyring3

groxxda committed:
cogl: 1.22.0 -> 1.22.2, enable wayland support

groxxda committed:
gnome3_22: init at 3.22

copied from 3.20
removed whitespaces at the end of lines
updated to 3.22 versions
removed webkitgtk24 where possible

gnome3_22.evolution: nspr include dir broken / not needed anymore
gnome3_22.gfbgraph: 0.2.2 -> 0.2.3
gnome3_22.gjs: 1.44.0 -> 1.46.0
gnome3_22.gnome-autoar: init at 0.1.1
gnome3_22.gnome-core.folks: 0.11.0 -> 0.11.2
gnome3_22.gnome-user-share 3.18.1 -> 3.18.3
gnome3_22.gnome-weather: 3.20.1 -> 3.20.2
gnome3_22.grilo: 0.3.1 -> 0.3.2
gnome3_22.grilo-plugins: 0.3.2 -> 0.3.3
gnome3_22.libgepub: init at 0.4
gnome3_22.libgweather: 3.20.1 -> 3.20.3
gnome3_22.rest: 0.7.93 -> 0.8.0
gnome3_22.totem: x86 patch is upstream
gnome3_22.tracker: 1.9.0 -> 1.10.0

groxxda committed:
gcab: 0.6 -> 0.7

groxxda committed:
libgtop: 2.32.0 -> 2.34.1

groxxda committed:
gvfs: use gnome3 as default

groxxda committed:
gtkmm_3: 3.20.0 -> 3.22.0

groxxda committed:
gnome3_22.vte_290: remove

groxxda committed:
pygobject: 3.20.0 -> 3.22.0

groxxda committed:
gnome3_22.vte-select-text: remove

groxxda committed:
gnome3_22.vte: remove selectTextPatch

groxxda committed:
gnome3_22.vte: 0.44.2 -> 0.46.0

groxxda commented:
glib: 03f1f7a is already in staging.  6ed55c0 and 7f3221f relate to the update and should be picked to staging independently

DamienCassou commented:
I'm building it. Please pay attention to https://github.com/NixOS/nixpkgs/pull/19099.

groxxda committed:
gnome3.gnome-terminal-server: init module

since gnome3.22 gnome-terminal uses a systemd service for it's backend
server.

groxxda commented:
@DamienCassou i cherry-picked the gnome-maps fix. but it does not work yet. (cannot find GWeather)

gnome-weather fails (cannot find GnomeDesktop)

gnome-terminal should be fixed in https://github.com/NixOS/nixpkgs/pull/19081/commits/1e28c6c5326d6d8036cbee849bbad935c4307f60

groxxda committed:
gnome3.evolution-data-server module: add self to systemd path

groxxda committed:
gnome3.tracker module: add self to systemd path

fixes GDBus.Error:org.freedesktop.systemd1.NoSuchUnit: Unit
tracker-store.service not found. - indexing for (null) will not work

groxxda committed:
gnome3.gvfs module: make systemd services available

DamienCassou commented:
I tried it this morning after a full night to compile. After the reboot, the computer stopped at a tty. `systemctl` told me that there is no `display-manager` unit.

groxxda commented:
@DamienCassou that's very unfortunate but I don't see how this could happen since there were no changes to the modules. Did you change anything in your working configuration.nix apart from `environment.gnome.packageSet = pkgs.gnome3_22;`?

groxxda committed:
gnome3_22.gnome-terminal: substitute gnome-shell dbus file path

instead of copying the file. this avoids name conflicts after the build.
Alternatively we could remove the file after building.

groxxda committed:
displayManager.gdm: only gnome-session needs to be on path

groxxda committed:
gnome3_22.mutter: add path to xwayland

groxxda committed:
gnome3_22.mutter: enable parallel build

groxxda committed:
gnome3_22.gnome-shell: enable parallel build

groxxda committed:
gnome3_22.gnome-terminal: enable parallel build

groxxda committed:
gnome3_22.gdm: enable parallel build

groxxda committed:
gnome3_22.gnome-session: add path to org.gnome.Shell.desktop

groxxda committed:
gnome3_22.gnome-session: lookup of gnome-settings-daemon.desktop

groxxda committed:
patch gnome-session path in desktop file

groxxda committed:
gnome-session: enable debug

groxxda committed:
xserverArgs fuckup

gnome-x-session provides good defaults which we really should not
override.
We have to add assertions to gdm.nix if the user specified one of those.

enableTCP must be configured through a gnome setting

dunno why we have terminate but it probably breaks stuff

We should expose configFile so we can use it from gdm module.

groxxda committed:
gdm module: enable user dbus session

groxxda committed:
gdm module: only make xserver args overrideable

groxxda committed:
gnome-themes-standard: -> 3.22.1

groxxda committed:
gdm: does not understand with-systemd

groxxda committed:
gdm: remove broken postinstall

groxxda committed:
evolutiondataserver parallel building

groxxda committed:
nspr fix

groxxda committed:
folks: -> 0.11.3

groxxda committed:
libgdata: 0.17.4 -> 0.17.6

groxxda commented:
I just rebased the PR and added more commits.

### gdm 3.22 works

[Here's a gist](https://gist.github.com/groxxda/9628b117168c37393415fd67d5876d81) with some screenshots:
- sd-pre-login shows that before login, no X is running. GDM 3.22 defaults to wayland. This may pose problems for users that rely on special X config, so should change this?
- sd-post-login shows that after user `test` logged on, a X server is started on a new vt, running as user `test`.
- sd-post-login-tty2 shows the vt that is switched to after `test` logged in. It runs on a new vt
- sd-post-login-tty1 shows that gdm will stay on vt1 after `test` logged and wait for other logins


Our current xserver module system is not entirely compatible with this behavor. For example it allows the user to set the tty to a specific value. This is probably not appropriate for sessions with a display manager.

See https://github.com/NixOS/nixpkgs/pull/19081/commits/7d29e74d3efd23741d97671d19ec8f0fd15c4611 for the current dirty hack.


Again ping gnome maintainers @lethalman @DamienCassou

DamienCassou commented:
I'm trying again, but my computer first needs to build it.

groxxda commented:
@DamienCassou do you want me to rebase onto staging? A few packages have been built there already

DamienCassou commented:
Alexander Ried <notifications@github.com> writes:
> @DamienCassou do you want me to rebase onto staging? A few packages have been built there already

do what you think is best, I will adapt. Just know that the longest it
takes to build, the less I can test.

-- 
Damien Cassou
http://damiencassou.seasidehosting.st

"Success is the ability to go from one failure to another without
losing enthusiasm." --Winston Churchill

DamienCassou commented:
Ok, my computer is now running gnome 3.22. I will report after a few hours of use.

DamienCassou commented:
First problem: I'm using the nixos module 'Emacs'  (systemd service for emacs daemon) and the client won't start:

```
$ emacsclient -c .bashrc 
Waiting for Emacs...
*ERROR*: Display :0 can’t be opened
```

vcunat commented:
I'm sorry to be so late. I read [fontconfig changelog](https://cgit.freedesktop.org/fontconfig/tree/README?id=2.12.1) and see no indication to need to bump `configVersion`. We only had the problem once, over the last few years I follow. Best leave it as it was, at least until someone finds (a serious) config incompatibility.

groxxda commented:
@vcunat thanks for the feedback, i'll file a pr for the fontconfig update
@DamienCassou i just tried to reproduce:
- set `services.emacs.enable = true;` in `configuration.nix`
- rebuild
- `systemctl --user start emacs`
- `emacsclient -c ~/.bashrc`

and emacs opens and shows the file just fine...

DamienCassou commented:
When starting plain Emacs, I get

```
** (emacs:2953): WARNING **: Error retrieving accessibility bus address: org.freedesktop.systemd1.NoSuchUnit: Unit at-spi-dbus-bus.service not found.
```

groxxda commented:
That's unrelated, accessibility bus is not strictly required. 
I'm aware and have already looked into it, adding a correct `--with-dbus-services=` to the build file and adding the at-spi2-core package to systemd.packages should fix it.
The patch somehow got lost in my rebases...

DamienCassou commented:
@groxxda I vote for merging that awesome work today and adding you as a Gnome maintainer. There will be time later to fix the problems we find.

groxxda commented:
Since many updates hit master and gnome already released a few minor updates, I rebased and force pushed.

fpletz commented:
Pushed some of the library updates to staging.

vcunat commented:
Fixed build of gvfs and gtkmm2 on staging.

groxxda commented:
Thanks @vcunat 
i only checked gvfs with gnome support and probably didn't check gtkmm2 at all..? :blush:

vcunat commented:
gtkmm2 had an interaction with a commit that's most likely not reachable from your PR.

groxxda commented:
Thanks to @fpletz the list of global updates in this PR is nearly finished. :beers:

Maybe someone wants to help out with the remaining issues..
- at-spi2 needs a module update. I'm not sure it was working before. But https://github.com/NixOS/nixpkgs/pull/19081#issuecomment-252007788 probably has all needed information
- gstreamer is a development version. We should either drop the update or wait for a release. Thoughts?
- gcab should just be merged. @fpletz we missed that somehow
- gtop should just be merged.
- cogl should be merged as is. In #18846 @lethalman requested the wayland support to be optional. But since everything else supports wayland unconditionally this is really a seperate issue imo.
- vte: i drop support for vte-ng in this pr. @cstrahan can you look into this, maybe it should just be a seperate expression..?

Commit "xserverArgs fuckup" is a blocker. We have to come up with a solution for the problem. I think what our xserver module does is just wrong for multi-user systems and calls for a breaking change. But we can probably also keep the broken behavior and make it a non-breaking change with mkDefault and so on..... This certainly needs feedback. Also note that this PR uses wayland for gdm although the module system suggests it uses X. We can probably use X for gdm but I see little value.

vcunat commented:
- gstreamer: so gnome 3.22 won't work with 1.8.x? Release might happen soon, as they're RC1 for two weeks already, but there might (theoretically) be lots of work on other packages to use the newer version.
- vte-select-text: it seemed pointless to me to create a separate expression when almost all would be the same. 45bf78373e

groxxda commented:
@vcunat I didn't test with 1.8 but i'm quite confident it works.
The separate expression could use override to share the common stuff.
I think that the patched version should not live inside of `pkgs/desktops/gnome` since it's not part of the gnome release..

@vcunat do you have an opinion on the xserver module?

vcunat commented:
- gstreamer: then I think it will be better to leave gstreamer changes to later, independently of gnome.
- vte: it could be done by an override, perhaps near the termite package. The point of the patch is that upstream refuse to expose the API that termite needs.
- xserver: I know of no particular use case for changing these options (but there may be some users of those around nixos.org). I know too little about all that &ndash; I didn't have an idea that the display manager could add some options in there...

vcunat commented:
Generally, how can I do changes in the gnome-3.20 expressions in a way that they won't be missing in gnome-3.22 ones and later? /cc @lethalman?

groxxda commented:
@vcunat i dont think there's an automated way.
This is how I do manual forward porting of the changes:

Assume I copied gnome3.20 to gnome3.22 at commit COPY
```
git format-patch COPY..HEAD -- pkgs/desktops/gnome-3/3.20
sed -i 's,gnome-3/3.20,gnome-3/3.22' *.patch
git am < *.patch
```

groxxda commented:
By the way I'm a bit disappointed that this PR is stalling.
I know that I touched many core packages and couldn't test them extensively (with different configurations) and individually. That's just impossible for one single person to do and since there are no binaries for this branch available no one can quickly test it.

vcunat commented:
In general, big PRs are harder to review, so they get less attention and progress slower than if split into multiple smaller ones.

At a quick glance I see only gnome3-specific packages affected now. Is that so?

vcunat commented:
If yes, and you have used the resulting gnome for a bit, I see no reason not to merge to master. (Except for a forward-porting pass, but that might better be postponed to 3.20 removal.)

groxxda commented:
The updates ordered before "init gnome 3.22" are globally reachable and will **at least** affect gnome3.20 which I didn't test at all and might affect other desktop stuff.
In particular _pygobject_ and _gtkmm_'s versioning suggest they correlate to the gnome release.

The big problem is that the modules need updates that are probably not compatible with gnome3.20.
Merging the gnome3.22 packages would be a good first step nevertheless to get binaries on hydra, maybe someone who was afraid of compiling everything will want to test the module updates then.

That being said, I have been using this branch since I filed the PR without real issues.
I only made one big change since then and that is the update of GDM (from 3.14 (**!**) to the current version) that I left out initially.
I occasionally pushed minor version updates that upstream released and that's about it.


This PR suggests that interest in gnome is very low, so I might as move my efforts to another topic.

DamienCassou commented:
@groxxda Gnome is quite a large beast. Thank you very much for your efforts. I think it's important to have recent Gnome versions in Linux distributions. You way want to add yourself to the list of Gnome maintainers. Thanks again.

vcunat commented:
Hmm, it's still a gtk3 rebuild and Hydra is relatively loaded, so I merged it to staging. That will give a little more time to test before master with some binaries available.

lethalman commented:
I think several people are using gnome in nixos nowadays. Not everyone has
the energy to look into this pr. Doesn't mean there is low interest.

On Thursday 20 October 2016, Alexander Ried <notifications@github.com>
wrote:

> The updates ordered before "init gnome 3.22" are globally reachable and
> will *at least* affect gnome3.20 which I didn't test at all and might
> affect other desktop stuff.
> In particular *pygobject* and *gtkmm*'s versioning suggest they correlate
> to the gnome release.
>
> The big problem is that the modules need updates that are probably not
> compatible with gnome3.20.
> Merging the gnome3.22 packages would be a good first step nevertheless to
> get binaries on hydra, maybe someone who was afraid of compiling everything
> will want to test the module updates then.
>
> That being said, I have been using this branch since I filed the PR
> without real issues.
> I only made one big change since then and that is the update of GDM (from
> 3.14 (*!*) to the current version) that I left out initially.
> I occasionally pushed minor version updates that upstream released and
> that's about it.
>
> This PR suggests that interest in gnome is very low, so I might as move my
> efforts to another topic.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/19081#issuecomment-255182390>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAdWmMvbYZvIW_TpOESVq03AcMsWFBK8ks5q161cgaJpZM4KKFp0>
> .
>


-- 
NixOS Linux <http://nixos.org>

lethalman commented:
Does it work? Did you guy test it? Merge :)

On Thu, Oct 20, 2016 at 6:02 PM, Vladimír Čunát <notifications@github.com>
wrote:

> If yes, and you have used the resulting gnome for a bit, I see no reason
> not to merge to master. (Except for a forward-porting pass, but that might
> better be postponed to 3.20 removal.)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/19081#issuecomment-255165928>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAdWmMs_UNRbf3NpQiQu0t-IDmUDs8v9ks5q156WgaJpZM4KKFp0>
> .
>



-- 
NixOS Linux <http://nixos.org>

vcunat commented:
@lethalman: it was merged to staging, as I wrote in my previous post. Hydra hasn't finished rebuilding yet, due to some previous unrelated evaluation errors, but staging might be mergeable in about a day, if there aren't many build regressions.

I'm briefly testing my system now, but that doesn't use gnome3 except a couple apps.
